### PR TITLE
Flush the gclog after GC

### PR DESCRIPTION
**Pull request by cshung**
This will allow us to see the log sooner while the process is still executing.

### CODE REVIEW

**Comment by Maoni0** [comment]
if you want to check this in you should make it more generic - I just did it as a temporary change (for example, in the scenarios I've been looking at I was happy with logging the first 6000 GCs). you could add a couple of more configs to specify the start/end GC indices. or by default not have the check for 6000 GCs.

**Review by Maoni0 — APPROVED** [review]
:shipit:

**Comment by Maoni0** [comment]
(actually as we talked about please add this to FATAL_GC_ERROR too)

**Comment by cshung** [comment]
I accidentally left `TRACE_GC` and `SIMPLE_DPRINTF` on during the last push. While we are at it, I want to take a look at what happens in the CI if they are left on. I will turn it off when I am done.

File: src/coreclr/gc/gcpriv.h [hunk 1]
```diff
@@ -108,8 +108,18 @@
 #pragma warning(disable:4477)
 #endif //_MSC_VER
 
+//#define TRACE_GC
+//#define SIMPLE_DPRINTF
+
+#if defined(TRACE_GC)
```
Comment (Maoni0) [inline src/coreclr/gc/gcpriv.h:114]: >#if defined [](http://example.com/codeflow?start=0&length=11)

let's do the SIMPLE_DPRINTF here too

File: src/coreclr/gc/gcpriv.h [hunk 2]
```diff
@@ -108,8 +108,18 @@
...
+void flush_gc_log (bool);
+#endif
+
 inline void FATAL_GC_ERROR()
 {
+#if defined(TRACE_GC)
```
Comment (Maoni0) [inline src/coreclr/gc/gcpriv.h:120]: same here

File: src/coreclr/gc/gcpriv.h [hunk 3]
```diff
@@ -913,8 +919,8 @@ class allocator
     void unlink_item_no_undo_added (unsigned int bn, uint8_t* item, uint8_t* previous_item);
 #if defined(MULTIPLE_HEAPS) && defined(USE_REGIONS)
     void count_items (gc_heap* this_hp, size_t* fl_items_count, size_t* fl_items_for_oh_count);
-    void rethread_items (size_t* num_total_fl_items, 
-                         size_t* num_total_fl_items_rethread, 
+    void rethread_items (size_t* num_total_fl_items,
```
Comment (Maoni0) [inline src/coreclr/gc/gcpriv.h:922]: unintentionally changes?

Comment (cshung) [reply src/coreclr/gc/gcpriv.h:922]: Oh yes, VSCode trims my end spaces again ...

**Pull request merged**